### ISSUE TITLE
RuntimeError when running main_discover.py with specific arguments

### ISSUE THREAD

**Issue by summeryeah** [LuigiRiz/NOPS#2]
Hi,

I'm currently running your project and I must say it's excellent work!

However, I ran into a RuntimeError when executing the following command:

`python main_discover.py -s 0 --dataset SemanticPOSS --dataset_config /segmentation/NOPS/config/semposs_dataset.yaml --num_heads=5 --overcluster_factor=3 --use_scheduler --adapting_epsilon_sk --use_uncertainty_queue --use_uncertainty_loss --uncertainty_percentile=0.3`

The error trace is as follows:

`File "/segmentation/NOPS/modules/Discoverer.py", line 359, in training_step
pred_sk = self.sk(
File "/ENVNOPS/lib/python3.8/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
return forward_call(*input, **kwargs)
File "/ENVNOPS/lib/python3.8/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
return func(*args, **kwargs)
File "/segmentation/NOPS/utils/sinkhorn_knopp.py", line 23, in forward
Q = torch.exp(logits / self.epsilon).t()
RuntimeError: t() expects a tensor with <= 2 dimensions, but self is 3D.
`
`logits: torch.Size([1, 192589, 4])`

The error seems to occur when the t() function expects a tensor with less than or equal to 2 dimensions, but receives a 3D tensor. I'd appreciate any help or insight you could provide on this issue.

Thanks for your time.

**Comment by LuigiRiz** [comment]
Hi, 

we recently used the code of this repo for another project and it didn't gave us any problem. I also tried to run the code with the same flags you used and I still get no error (in my case 'logits' at line 22 of "NOPS/utils/sinkhorn_knopp.py" has torch.Size([136442, 4]) ).

Please let me know if you have any further questions or if there's anything else I can do to help.

**Comment by summeryeah** [comment]
Thank you very much for your quick response! Your help has been invaluable, and I will revisit the entire process to make sure everything is in order.

I have another small question. I want to confirm about line 53 in main_discover.py:

`parser.add_argument("--pretrained", type=str, help="pretrained checkpoint path")
`

Does the "pretrained checkpoint path" refer to the "checkpoint dir" in line 16 of main_pretrain.py:

`parser.add_argument("--checkpoint_dir", default="checkpoints_pretraining", type=str, help="checkpoint dir")
`

In other words, is it correct to modify line 53 in main_discover.py to::

`parser.add_argument("--pretrained", default="/segmentation/NOPS/checkpoints_pretraining/pretrain-0-SemanticKITTI-Jun06_06-10-14.ckpt", type=str, help="pretrained checkpoint path")
`
Looking forward to your confirmation.

Thank you again for your time and assistance.

**Comment by LuigiRiz** [comment]
Hi,
yes, line 53 refers to the checkpoints one can obtain through the main_pretrain.py script. However, as also reported in the paper, I strongly suggest to run just the main_discover.py script that jointly learns base and novel classes, as the performances in this last case are much higher.

Hope this can help you

**Comment by summeryeah** [comment]
Thank you for your confirmation and advice. I'll try using just main_discover.py. I appreciate your help!
